ios/AppLovinMAX.m

```diff
@@ -1327,6 +1319,81 @@ - (void)log:(NSString *)format, ...
     NSLog(@"[%@] [%@] %@", SDK_TAG, TAG, message);
 }
 
+#pragma mark - Ad Info
+
+- (NSDictionary<NSString *, id> *)adInfoForAd:(MAAd *)ad
+{
+    return @{@"adUnitId" : ad.adUnitIdentifier,
+             @"creativeId" : ad.creativeIdentifier ?: @"",
+             @"networkName" : ad.networkName,
+             @"placement" : ad.placement ?: @"",
+             @"revenue" : @(ad.revenue),
+             @"waterfall": [self createAdWaterfallInfo: ad.waterfall]};
+}
+
+#pragma mark - Waterfall Information
+
+- (NSDictionary<NSString *, id> *)createAdWaterfallInfo:(MAAdWaterfallInfo *)waterfallInfo
```

Comment:
Can we try to match Unity one-to-one? I think the `if ( !waterfallInfo ) return waterfallInfoDict;` is not necessary but we try to match the code as much as possible

https://www.diffchecker.com/erjOqvhM

Reply:
Updated more close to the Unity implementation.   The differences are now:

1. BundleUtils.toJSONObject: implemented the equivalent function
2. Output type:  Unity converts all fields to string. 

Please review.   Also, this has some fix for my bad.   I ran indent.